data/locations.js

```diff
@@ -0,0 +1,131 @@
+import db from '../config/mongoCollections.js';
+import validators from '../validation.js';
+
+const _validateZipcode = (zipcode) => {
+    if (zipcode.length < 5 || zipcode.length > 10) {
+        throw "Invalid zipcode length";
+    }
+    return zipcode;
+}
+
+const _validateLatLong = (value, name) => {
+        if (typeof value !== 'number') {  
+        throw `Invalid ${name} value`;  
+    }  
+    if (name === "Latitude") {  
+        if (value < -90 || value > 90) {  
+            throw `Invalid ${name} value, must be between -90 and 90`;  
+        }  
+    } else if (name === "Longitude") {  
+        if (value < -180 || value > 180) {  
+            throw `Invalid ${name} value, must be between -180 and 180`;  
+        }  
+    } else {  
+        throw `Unknown coordinate name: ${name}`;  
+    }
+    return value;  
+}
+
+const _validateStateCode = (state_code) => {
+    if (state_code.length !== 2) {
+        throw "State code must be 2 characters";
+    }
+    return state_code.toUpperCase();
+}
+
+const _validateRadius = (radius) => {
+    if (typeof radius !== 'number' || radius <= 0) {
+        throw "Invalid radius value";
+    }
+    return radius;
+}
+
+const _validatePositiveNumber = (value, name) => {
+    if (typeof value !== 'number' || value < 0) {
```

Comment:
Try this, but you may want to add more checks if you want floats to be valid too

```suggestion
    if (Number.isNaN(value) || value < 0) {
```